### PR TITLE
Add option to only build library

### PR DESCRIPTION
**Pull request by makaimann**
This PR would add an option to only build the CVC4 library and not the parser or executable. This can be used for projects that only intend to use CVC4 through the API.

It seems to be working now, but it's not necessarily the cleanest solution. In particular, if you'd like the polarity to be different I'm happy to change that. Polarity meaning something like `"${WITH_BINARY}" STREQUAL "YES"` instead of `NOT "${LIB_ONLY} STREQUAL "YES"` which is admittedly a little strange.

### CODE REVIEW

File: configure.sh [hunk 1]
```diff
@@ -334,6 +339,8 @@ do
     --symfpu-dir) die "missing argument to $1 (try -h)" ;;
     --symfpu-dir=*) symfpu_dir=${1##*=} ;;
 
+    --lib-only) lib_only=YES ;;
```
Comment (mpreiner) [inline configure.sh:342]: Use ON/OFF similar to the other options.

File: src/CMakeLists.txt [hunk 2]
```diff
@@ -843,7 +843,10 @@ set(KINDS_FILES
 add_subdirectory(base)
 add_subdirectory(expr)
 add_subdirectory(options)
-add_subdirectory(parser)
+if (NOT "${LIB_ONLY}" STREQUAL "YES")
+  message("Adding parser")
```
Comment (mpreiner) [inline src/CMakeLists.txt:847]: I don't think that we need this message.

File: src/CMakeLists.txt [hunk 3]
```diff
@@ -843,7 +843,10 @@ set(KINDS_FILES
 add_subdirectory(base)
 add_subdirectory(expr)
 add_subdirectory(options)
-add_subdirectory(parser)
+if (NOT "${LIB_ONLY}" STREQUAL "YES")
```
Comment (mpreiner) [inline src/CMakeLists.txt:846]:
```suggestion
if (NOT LIB_ONLY)
```

Should also work.

File: src/CMakeLists.txt [hunk 4]
```diff
@@ -935,7 +938,10 @@ target_link_libraries(cvc4 ${RT_LIBRARIES})
 # target_link_libraries(...) with object libraries for cmake versions <= 3.12.
 # Thus, we can only visit main as soon as all dependencies for cvc4 are set up.
 
-add_subdirectory(main)
+if (NOT "${LIB_ONLY}" STREQUAL "YES")
```
Comment (mpreiner) [inline src/CMakeLists.txt:941]:
```suggestion
if (NOT LIB_ONLY)
```

File: src/CMakeLists.txt [hunk 5]
```diff
@@ -935,7 +938,10 @@ target_link_libraries(cvc4 ${RT_LIBRARIES})
...
+  message("Adding main")
```
Comment (mpreiner) [inline src/CMakeLists.txt:942]: Ditto

File: test/CMakeLists.txt [hunk 6]
```diff
@@ -21,7 +21,9 @@ add_custom_target(check
 #-----------------------------------------------------------------------------#
 # Add subdirectories
 
-add_subdirectory(regress)
+if (NOT "${LIB_ONLY}" STREQUAL "YES")
```
Comment (mpreiner) [inline test/CMakeLists.txt:24]:
```suggestion
if (NOT LIB_ONLY)
```

**Comment by makaimann** [comment]
Thanks for the review! Addressed your comments in the last few commits.

File: configure.sh [hunk 7]
```diff
@@ -80,6 +80,9 @@ Optional Path to Optional Packages:
   --poly-dir=PATH          path to top level of LibPoly source tree
   --symfpu-dir=PATH        path to top level of SymFPU source tree
 
+Build limitations:
+  --lib-only               only build the library, but not the executable or the parser (default: off)
```
Comment (mpreiner) [inline configure.sh:84]: Make sure this line is 80 chars.

File: src/CMakeLists.txt (hunk 5 above)
Comment (mpreiner) [reply src/CMakeLists.txt:941]: This message can be removed.

File: configure.sh (hunk 7 above)
Comment (makaimann) [reply configure.sh]: Thanks -- done in 085bf094d918953c803d8b76848996ae4178445e

**Comment by makaimann** [comment]
> `BUILD_LIB_ONLY` should be initialized in CMakeLists.txt similar to `USE_PYTHON3`.

I made this change in 8d59ae83bb24d5e886f487d8a8355269c22af8ba. I also added it in the printed configuration, but don't feel strongly about it being there or not.

**Pull request merged**